modules/common/service/service.go

```diff
@@ -105,6 +150,18 @@ func (s *Service) CreateOrPatch(
 		h.GetLogger().Info(fmt.Sprintf("Service %s - %s", service.Name, op))
 	}
 
+	// update the service instance with the ip/host information
+	s.clusterIPs = service.Spec.ClusterIPs
+	s.serviceHostname = fmt.Sprintf("%s.%s.svc", service.Name, service.GetNamespace())
+
+	if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
+		if len(service.Status.LoadBalancer.Ingress) > 0 {
```

Comment:
Can it take some time for the service status to update?

Reply:
it probably could, right now we are not using GetExternalIPs(). if we see there are issues, GetExternalIPs() could be updated to query the service.